### ISSUE TITLE
jacoco-maven-plugin has dependency on Maven 3.0 which has a known security issue.

### ISSUE THREAD

**Issue by rcollette** [jacoco/jacoco#1203]
### Scenario

* JaCoCo version: 0.8.7
* Operating system: Any
* Tool integration: Maven 
* Description of your use case: Security

### Current Behaviour
BlackDuck reports a transitive dependency on Maven 3.0 which has a security issue.

![image](https://user-images.githubusercontent.com/136476/123120101-0dbe3080-d412-11eb-8f74-7585af836d4a.png)

### Wanted Behaviour
Update the version of Maven referenced to 3.8.1

**Comment by marchof** [comment]
@rcollette Actually JaCoCo can be used with any Maven version > 3.0. We do not want to force our users to install a specific version.

@Godin I just checked the dependencies for the Maven plugin and wonder whether we use the correct scope. Shouldn't we better use `provided`? At least the [example in the Maven documentation](https://maven.apache.org/guides/plugin/guide-java-plugin-development.html) does.

```
    <dependency>
      <groupId>org.apache.maven</groupId>
      <artifactId>maven-plugin-api</artifactId>
      <scope>provided</scope>
      <version>${project.prerequisites.maven}</version>
    </dependency>
    <dependency>
      <groupId>org.apache.maven</groupId>
      <artifactId>maven-core</artifactId>
      <scope>provided</scope>
      <version>${project.prerequisites.maven}</version>
    </dependency>
```

**Comment by Godin** [comment]
Version of `maven-core` dependency is about minimally required Maven version, actual version used at runtime is a version of Maven.

> We do not want to force our users to install a specific version.

I agree - weird to ask plugins to update minimally required version because of changes in Maven unrelated to the plugin - if somebody wanna update Maven, he can just do it.

> checked the dependencies for the Maven plugin and wonder whether we use the correct scope. Shouldn't we better use provided?

I checked dependencies of Maven plugins such as [`maven-compiler-plugin`](https://maven.apache.org/plugins/maven-compiler-plugin/dependencies.html), [`maven-resources-plugin`](https://maven.apache.org/plugins/maven-resources-plugin/dependencies.html), [`maven-deploy-plugin`](https://maven.apache.org/plugins/maven-deploy-plugin/dependencies.html), [`maven-jar-plugin`](https://maven.apache.org/plugins/maven-jar-plugin/dependencies.html), [`maven-dependency-plugin`](https://maven.apache.org/plugins/maven-dependency-plugin/dependencies.html), [`maven-help-plugin`](https://maven.apache.org/plugins/maven-help-plugin/dependencies.html) and like in `jacoco-maven-plugin` they all for some reason declare dependency on `maven-core` with scope `compile`.

However

> At least the [example in the Maven documentation](https://maven.apache.org/guides/plugin/guide-java-plugin-development.html) does

also

```
mvn archetype:generate \
  -DgroupId=sample.plugin \
  -DartifactId=hello-maven-plugin \
  -DarchetypeGroupId=org.apache.maven.archetypes \
  -DarchetypeArtifactId=maven-archetype-plugin
```

generates plugin that uses `provided` scope - https://github.com/apache/maven-archetypes/blob/1beaa8755a2b2c69861d64364ccb88d92cab387a/maven-archetype-plugin/src/main/resources-filtered/archetype-resources/pom.xml#L45-L56

So indeed let's declares them as `provided` - #1205

However don't know if this will be enough to silence message from BlackDuck. Also wondering if similar ticket gonna be created for all the Maven plugins out there 😆

**Comment by patpatpat123** [comment]
Hello Team, just to confirm, this is unfortunately not enough to silence BlackDuck CVE flag 🥲

Thank you for the work and thank you for this great plugin!

**Comment by patpatpat123** [comment]
Just asking, if you allow me, " if somebody wanna update Maven, he can just do it." I would like to fix this CVE, and if this requires me to update anything, I am more than happy to do it and consequences will be mine only.

With that said, may I ask how do I do so please?

Thank you

**Comment by Godin** [comment]
@patpatpat123 

> this is unfortunately not enough to silence BlackDuck CVE flag

Does this mean that using BlackDuck you scanned not yet released JaCoCo version `0.8.8-SNAPSHOT` where scope of dependency was changed to `provided`?

If so, then please report to BlackDuck, because they are wrong with their detection of CVE-2021-26291 in JaCoCo - quoting https://github.com/jacoco/jacoco/issues/1203#issuecomment-868987460

> Version of maven-core dependency is about minimally required Maven version, **actual version used at runtime is a version of Maven**.

and so answering on

> I would like to fix this CVE, and if this requires me to update anything, I am more than happy to do it and consequences will be mine only.
>
> With that said, may I ask how do I do so please?

you just have to **use Maven version `3.8.1` or greater** and that's it.

**Comment by patpatpat123** [comment]
Hello Team,

Just wanted to provide the information, version 0.8.8 does indeed fix the CVE.

Thank you for the help and good day!
